VespaGAS/Managers/OilMixCalculator.swift

```diff
@@ -0,0 +1,17 @@
+import Foundation
+
+typealias GasType = Float
+typealias OilType = Float
+typealias OilMixType = Float
+
+let MLInLiter: Float = 1000
+
+struct OilMixCalculator {
+    static func oilMLForGasoline(gasInLiter: GasType, oilPercentage: OilMixType) -> OilType? {
+        if oilPercentage == 0 {
+            return nil
```

Comment:
In terms of Domain Design, I think this might make better sense that this returns a non-Optional `OilType` and this branch returns `0`. If you think about it, you don't add a `nil` amount of oil to an engine, you'd add 0ml :) 

this may be a good book if you're interested in domain driven design - it is very short and I highly recommend reading it https://www.infoq.com/minibooks/domain-driven-design-quickly